### PR TITLE
Docs redesign: Question component

### PR DESCRIPTION
**Pull request by st-jay**
## Summary of change
- Updated the question component
- Updated the `snippetConfigForm/formQuestion` component used in passwordless docs

### Design references
- Design: https://zpl.io/adQE8yK
- Other updates
  - Updated the icons in left nav ( e.g. icons for Netlify, Redwood, etc. )
    - `height: 16px`
    - `width: auto`
    - Space between the link's text and icon should be 8px. The link's text has a `padding-right: 16px`, so added `margin-left: -8px` to icon.
  - Removed `margin-top` for first image that appears inside a tabs container.
  - The `You can refresh the page to undo your selection` message could overlap the answers in the Question component. If the answers overlap the message, the method in `question/index.tsx`, `updateAnswerBoxPaddingIfAnswersOverflow` add extra space to `padding-bottom` of the container which prevents this overlap. It is not needed in the `snippetConfigForm/formQuestion` component as it does not have the message. 

## Related issues
- https://github.com/supertokens/main-website/issues/576
- #257 

## Checklist
- [ ] ~Algolia search needs to be updated? (If there is a new sub docs project, then yes)~
- [ ] ~Sitemap needs to be updated? (If there is a new sub docs project, then yes)~
- [ ] ~Checked for broken links? (Run `cd v2 && MODE=production npx docusaurus build`)~
- [ ] ~Changes required to the demo apps corresponding to the docs?~

## Remaining TODOs for this PR
- [x] Design review

### CODE REVIEW

File: v2/src/components/question/index.tsx [hunk 1]
```diff
@@ -109,43 +71,27 @@ export function Question(props: PropsWithChildren<{
...
+        );
     }
 }
 
 type AnswerProps = {
     title: string,
-    onClick?: () => void
-}
+    onClick?: () => void,
+    isLast: boolean
```
Comment (nkshah2) [inline v2/src/components/question/index.tsx:81]: What is this used for?

**Comment by nkshah2** [comment]
Screenshot 2022-03-23 at 11 48 11 AM

Screenshot 2022-03-23 at 11 48 04 AM

I think the default question component added some spacing above and below it

File: v2/src/components/question/index.tsx [hunk 2]
```diff
@@ -1,86 +1,75 @@
...
+            const availableWidth = answerBoxWidth - messageWidth - 36;
+
+            let totalAnswersWidth = 0;
+            for (const answer of answersInsideContainer) {
+                const width = answer.getBoundingClientRect().width;
+                totalAnswersWidth = totalAnswersWidth + width + 24;
+            }
+            setExtraPaddingToAnswerBox(totalAnswersWidth > availableWidth);
+        }
+    };
```
Comment (nkshah2) [inline v2/src/components/question/index.tsx:37]: Why was this required? I thought this was just a design change

Comment (st-jay) [reply v2/src/components/question/index.tsx:37]: it was needed to be conditional, if the answers overlap the message, then only we add extra padding, otherwise we do not add extra padding

Comment (nkshah2) [reply v2/src/components/question/index.tsx:37]: What does overlap mean here?

Comment (nkshah2) [reply v2/src/components/question/index.tsx:37]: Also `const messageWidth = 227;` where did you get this number from? and how will this work on mobile?

Comment (st-jay) [reply v2/src/components/question/index.tsx:37]: on iPad Pro, the message `Refresh the page to undo your selection` overlaps the answer buttons.

I got the `const messageWidth = 227;` from the width of `Refresh the page to undo your selection`. Though I realise now it would be better to get the width on execution, otherwise we would need to update this value any time we change the style or content of that message. 

image

Comment (nkshah2) [reply v2/src/components/question/index.tsx:37]: Wouldnt it make more sense to give the text a fixed width based on its parent width instead? (and the same for the buttons container too)

Comment (nkshah2) [reply v2/src/components/question/index.tsx:37]: Also this happens if I open the preview on an actual iPad Pro

IMG_64B76BD3DEF6-1
